src/dotnet/commands/dotnet-sln/add/Program.cs

```diff
@@ -17,11 +19,14 @@ internal class AddProjectToSolutionCommand : CommandBase
         private readonly AppliedOption _appliedCommand;
         private readonly string _fileOrDirectory;
         private readonly bool _inRoot;
-        
+        private readonly string _relativeRoot;
```

Comment:
Perhaps we should just store `_relativeRootSolutionFolders`?

That is to say, in the constructor, using a local `relativeRoot` instead of the field:

```
 _relativeRootSolutionFolders = relativeRoot?.Split(Path.DirectorySeparatorChar);
```

And thus in `DetermineSolutionFolder`:

```
...
if (_relativeRootSolutionFolders != null)
{
    return _relativeRootSolutionFolders;
}
...
```

I think that would be a little bit cleaner than having `DetermineSolutionFolder` have side-effects (i.e. assignment to `_relativeRootSolutionFolders `).

What do you think?

Reply:
That _would_ make it a little cleaner. Is this the last change? :)

Reply:
I think so 😄 🤞 🎉